### ISSUE TITLE
`command` is an unwieldy function

### ISSUE THREAD

**Issue by dnathe4th** [octachrome/treason#35]
In my fork I have taken some time to refactor `command` to be a bit more manageable, with clearer mapping between valid commands and states, as well as deduplication of some of the validation-type checks in each function.

Would you entertain a refactor like this?
https://github.com/dnathe4th/treason/commit/5ce8493407e1b4e28a79e5a6f17acfba413da82e#diff-41c4530a42496f695607188d7bbe06d5

**Comment by octachrome** [comment]
First I feel the need to apologise for my code in this file. It is an ugly mess long overdue refactoring. Thanks for taking the time. If you are willing to create a pull request then I would have some comments/suggestions and we could get this merged. It would be good to include this commit too: https://github.com/dnathe4th/treason/commit/b5140124a2f772ccade866b7d224a744b098fbb7.

**Comment by dnathe4th** [comment]
Happily. Let me cherry-pick it off my fork after work today.

I'm new-ish to OSS licensing. I created an amendment in my fork for my
work, but how should that look once it gets merged into the main project?

**Comment by octachrome** [comment]
Good question. In your license amendments you have claimed copyright/ownership on your modifications, which is your right: https://softwareengineering.stackexchange.com/questions/211365/who-owns-modifications-to-open-source-software. I think I would prefer for you to transfer ownership of your contributions to me upon merging, to avoid losing my ability to do what I want with the project as a whole (otherwise I would have to get your permission on everything or revert your changes). I would like to add a list of contributors to the license file, where you would be credited. How do you feel about that?

**Comment by dnathe4th** [comment]
That sounds good to me

**Comment by dnathe4th** [comment]
As long as you're ok taking ownership of my bad code :- )

**Comment by octachrome** [comment]
Ok cool. If you wouldn't mind updating the license before merging then, remove your copyright and add yourself as a contributor instead. If you want to itemize your contributions that's fine.

**Comment by dnathe4th** [comment]
Yeah need to do a bunch of clean up, too much going on on my own branches.
Later today.
